src/python/WMComponent/ErrorHandler/ErrorHandlerPoller.py

```diff
@@ -71,7 +71,11 @@ def __init__(self, config):
             self.maxRetries = self.config.ErrorHandler.maxRetries
         else:
             self.reqAuxDB = ReqMgrAux(self.config.General.ReqMgr2ServiceURL)
-            self.maxRetries = self.reqAuxDB.getWMAgentConfig(self.config.Agent.hostName).get("MaxRetries")
+            agentConfig = self.reqAuxDB.getWMAgentConfig(self.config.Agent.hostName)
+            if agentConfig.get("UserDrainMode") and agentConfig.get("SpeedDrainMode"):
```

Comment:
SpeedDrainMode is set to true when any one of those 3 metrics goes Enabled. So we also need to check for the SpeedDrainConfig/NoJobRetries/Enabled value.